src/vstest.console/Processors/EnableBlameArgumentProcessor.cs

```diff
@@ -220,9 +224,78 @@ public void Initialize(string argument)
                 }
             }
 
+            // Update run settings.
             runSettingsManager.UpdateRunSettingsNodeInnerXml(Constants.DataCollectionRunSettingsName, dataCollectionRunSettings.ToXml().InnerXml);
         }
 
+        /// <summary>
+        /// Get results directory.
+        /// </summary>
+        /// <param name="settings">Settings xml.</param>
+        /// <returns>Results directory.</returns>
+        private string GetResultsDirectory(string settings)
+        {
+            string resultsDirectory = null;
+            if (settings != null)
+            {
+                try
+                {
+                    RunConfiguration runConfiguration = XmlRunSettingsUtilities.GetRunConfigurationNode(settings);
+                    resultsDirectory = RunSettingsUtilities.GetTestResultsDirectory(runConfiguration);
+                }
+                catch (SettingsException se)
+                {
+                    if (EqtTrace.IsErrorEnabled)
+                    {
+                        EqtTrace.Error("EnableBlameArgumentProcessor: Unable to get the test results directory: Error {0}", se);
+                    }
+                }
+            }
+
+            return resultsDirectory;
+        }
+
+        /// <summary>
+        /// Checks if dump collection is supported.
+        /// </summary>
+        /// <returns>Dump collection supported flag.</returns>
+        private bool IsDumpCollectionSupported()
+        {
+            var dumpCollectionSupported = this.environment.OperatingSystem == PlatformOperatingSystem.Windows &&
+                    this.environment.Architecture != PlatformArchitecture.ARM64 &&
+                    this.environment.Architecture != PlatformArchitecture.ARM;
+
+            if (!dumpCollectionSupported)
+            {
+                Output.Warning(false, CommandLineResources.BlameCollectDumpNotSupportedForPlatform);
+            }
+
+            return dumpCollectionSupported;
+        }
+
+        /// <summary>
+        /// Check if collect dump key is valid.
+        /// </summary>
+        /// <param name="collectDumpKey">Collect dump key.</param>
+        /// <returns>Flag for collect dump key valid or not.</returns>
+        private bool ValidateCollectDumpKey(string collectDumpKey)
```

Comment:
I am confused. How is this part of this PR ?

Reply:
We have made the parsing logic generic with my PR as per Mayank's PR comment. Thus blame, diag and logger have been changed to use the new generic logic. 

These are just refactoring and not the code changes.